packages/grafana-ui/src/components/transitions/SlideOutTransition.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { css } from 'emotion';
+import { CSSTransition } from 'react-transition-group';
+import { stylesFactory } from '../../themes';
+
+const getStyles = stylesFactory((duration: number, measurement: 'width' | 'height', size: number) => {
+  return {
+    enter: css`
+      label: enter;
+      ${measurement}: 0;
+      opacity: 0;
+    `,
+    enterActive: css`
+      label: enterActive;
+      ${measurement}: ${size}px;
+      opacity: 1;
+      transition: opacity ${duration}ms ease-out, ${measurement} ${duration}ms ease-out;
+    `,
+    exit: css`
+      label: exit;
+      ${measurement}: ${size}px;
+      opacity: 1;
+    `,
+    exitActive: css`
+      label: exitActive;
+      opacity: 0;
+      ${measurement}: 0;
+      transition: opacity ${duration}ms ease-out, ${measurement} ${duration}ms ease-out;
+    `,
+  };
+});
+
+type Props = {
```

Comment:
Just of curiosity why a type and not an interface?

Reply:
Kinda unconscious personal preference (easier to type I guess). I rarely need to implement or extend `Props` so `type` seems fine there, not sure if there are any other benefits of using interface instead of type.